src/overlays/actors/ovl_En_Bigpo/z_en_bigpo.c

```diff
@@ -94,161 +152,1305 @@ static DamageTable D_80B65044 = {
+        gSPSegment(dispHead + 1, 0x0C, &D_801AEFA0);
+        gSPSegment(dispHead + 2, 0x08,
+                   Gfx_EnvColor(globalCtx->state.gfxCtx, this->mainColor.r, this->mainColor.g, this->mainColor.b,
+                                this->mainColor.a));
+        POLY_OPA_DISP = SkelAnime_Draw2(globalCtx, this->skelAnime.skeleton, this->skelAnime.limbDrawTbl,
+                                        EnBigpo_OverrideLimbDraw2, EnBigpo_PostLimbDraw2, &this->actor, dispHead + 3);
+
+    } else {
+        dispHead = POLY_XLU_DISP;
+        gSPDisplayList(dispHead, &sSetupDL[6 * 0x19]);
+        gSPSegment(dispHead + 1, 0x0C, &D_801AEF88);
+        gSPSegment(dispHead + 2, 0x08,
+                   Gfx_EnvColor(globalCtx->state.gfxCtx, this->mainColor.r, this->mainColor.g, this->mainColor.b,
+                                this->mainColor.a));
+        POLY_XLU_DISP = SkelAnime_Draw2(globalCtx, this->skelAnime.skeleton, this->skelAnime.limbDrawTbl,
+                                        EnBigpo_OverrideLimbDraw2, EnBigpo_PostLimbDraw2, &this->actor, dispHead + 3);
+    }
+
+    func_800BE680(globalCtx, &this->actor, this->limbPos, 9, this->actor.scale.x * 71.428566f * this->unk220, 0,
+                  this->unk21C, 0x14);
+
+    SysMatrix_SetCurrentState(&this->drawMtxF);
+    EnBigpo_DrawLantern(&this->actor, globalCtx);
+    if (this->actionFunc == EnBigpo_SpawnPoCutscene6) {
+        // call the other unk draw func
+        EnBigpo_Draw4(&this->actor, globalCtx);
+    }
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+void EnBigpo_DrawScoopSoul(Actor* thisx, GlobalContext* globalCtx) {
+    EnBigpo* this = (EnBigpo*)thisx;
+    s32 pad;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    func_8012C2DC(globalCtx->state.gfxCtx);
+
+    gSPSegment(POLY_XLU_DISP++, 0x08,
+               Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0, 0, 0x20, 0x40, 1, 0,
+                                (globalCtx->gameplayFrames * -0xF) & 0x1FF, 0x20, 0x80));
+
+    gDPSetPrimColor(POLY_XLU_DISP++, 0x80, 0x80, 0xFF, 0xFF, 0xAA, this->mainColor.a);
+
+    gDPSetEnvColor(POLY_XLU_DISP++, this->mainColor.r, this->mainColor.g, this->mainColor.b, 0xFF);
+
+    Lights_PointNoGlowSetInfo(&this->fires[0].info, this->actor.world.pos.x, this->actor.world.pos.y,
+                              this->actor.world.pos.z, this->mainColor.r, this->mainColor.g, this->mainColor.b,
+                              this->mainColor.a * 2);
+
+    Matrix_RotateY(BINANG_ROT180(func_800DFCDC(ACTIVE_CAM)), MTXMODE_APPLY);
+
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+    gSPDisplayList(POLY_XLU_DISP++, &D_06001BB0);
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+// this matches without OPENDISPS but with it has stack issues, might be able to find an alternative match with the
+// macros
+void EnBigpo_DrawLantern(Actor* thisx, GlobalContext* globalCtx) {
+    EnBigpo* this = (EnBigpo*)thisx;
```

Comment:
The more I've thought about it, the more important I think it is to find a match with the macros. The match as is is fine though, and leaving this comment here is perfect